apps/src/lib/ui/accounts/ManageLinkedAccounts.jsx

```diff
@@ -10,10 +10,53 @@ const OAUTH_PROVIDERS = {
   GOOGLE: 'google_oauth2',
   FACEBOOK: 'facebook',
   CLEVER: 'clever',
-  MICROSOFT: 'microsoft',
+  MICROSOFT: 'windowslive',
 };
+export const ENCRYPTED = `*** ${i18n.encrypted()} ***`;
 
 export default class ManageLinkedAccounts extends React.Component {
+  static propTypes = {
+    userType: PropTypes.string.isRequired,
+    authenticationOptions: PropTypes.arrayOf(PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      credential_type: PropTypes.string.isRequired,
+      email: PropTypes.string,
+      hashed_email: PropTypes.string,
+    })).isRequired,
+    connect: PropTypes.func.isRequired,
+    disconnect: PropTypes.func.isRequired,
+  };
+
+  getAuthenticationOption = (provider) => {
+    return this.props.authenticationOptions.find(option => {
+      return option.credential_type === provider;
+    });
+  };
+
+  getEmailForProvider = (provider) => {
+    const authOption = this.getAuthenticationOption(provider);
+    if (authOption) {
+      if (this.props.userType === 'student') {
+        return ENCRYPTED;
+      }
+      return authOption.email;
+    }
+  };
+
+  toggleProvider = (provider) => {
+    const authOption = this.getAuthenticationOption(provider);
+    if (authOption) {
+      this.props.disconnect(authOption.id).then(_, this.onFailure);
+    } else {
+      this.props.connect(provider);
+    }
+  };
+
+  onFailure = (error) => {
+    // TODO: (madelynkasula) display error to user
```

Comment:
i'm not really sure how we want to surface a `DELETE /users/auth/:id/disconnect` to users. some options:
- display a notification (via our React notification component)
- display the error message next to the "Connect/Disconnect" button
- other thoughts/options?

Reply:
I think either of those would work.  We don't have an intermediate state yet, but I'd be most likely to put the error wherever that 'in progress' status lives - maybe on the button itself, even.  Our desired outcome is probably for the user to click the disconnect button again, in case it's an intermittent issue.